src/backend/dx11/src/lib.rs

```diff
@@ -454,20 +449,86 @@ impl hal::Device<Backend> for Device {
         mut size: u64,
         usage: buffer::Usage,
     ) -> Result<UnboundBuffer, buffer::CreationError> {
-        unimplemented!()
+        use buffer::Usage;
+
+        let mut bind = 0;
+
+        if usage.contains(Usage::UNIFORM) { bind |= d3d11::D3D11_BIND_CONSTANT_BUFFER; }
+        if usage.contains(Usage::VERTEX) { bind |= d3d11::D3D11_BIND_VERTEX_BUFFER; }
+        if usage.contains(Usage::INDEX) { bind |= d3d11::D3D11_BIND_INDEX_BUFFER; }
+
+        // TODO: >=11.1
+        if usage.contains(Usage::UNIFORM_TEXEL) ||
+           usage.contains(Usage::STORAGE_TEXEL) { bind |= d3d11::D3D11_BIND_SHADER_RESOURCE; }
+
+        // TODO: how to do buffer copies
+        // if usage.contains(Usage::TRANSFER_SRC) ||
+        //    usage.contains(Usage::TRANSFER_DST) { bind |= d3d11::D3D11_BIND_UNORDERED_ACCESS; }
+
+        Ok(UnboundBuffer {
+            bind,
+            size,
+            requirements: memory::Requirements {
+                size,
+                alignment: 1,
+                type_mask: 0x7,
+            }
+        })
     }
 
     fn get_buffer_requirements(&self, buffer: &UnboundBuffer) -> memory::Requirements {
-        unimplemented!()
+        buffer.requirements
     }
 
     fn bind_buffer_memory(
         &self,
         memory: &Memory,
         offset: u64,
-        buffer: UnboundBuffer,
+        unbound_buffer: UnboundBuffer,
     ) -> Result<Buffer, device::BindError> {
-        unimplemented!()
+        // TODO: offset
+        assert_eq!(0, offset);
+        // TODO: structured buffers
+        assert_eq!(0, unbound_buffer.bind & d3d11::D3D11_BIND_SHADER_RESOURCE);
+        // TODO: change memory to be capable of more than one buffer?
+        // assert_eq!(None, memory.buffer);
+
+        let (usage, cpu) = match (memory.can_download(), memory.can_upload()) {
+            (true,  true)  => (d3d11::D3D11_USAGE_DYNAMIC, d3d11::D3D11_CPU_ACCESS_WRITE),
```

Comment:
A possible d3d11 limitation? Not being able to read from the upload buffer (even if not encouraged).